src/computation/StreamedComputation.cpp

```diff
@@ -65,22 +65,38 @@ void StreamedComputation::performComputation(std::shared_ptr<wrench::ActionExecu
         }
 
         // Process next blocks: compute block i while reading block i+i
+        size_t n_prefetch_blocks_counter = 0;
         for (int i=0; i < num_blocks - 1; i++) {
             double num_bytes = std::min<double>(SimpleSimulator::xrd_block_size, data_to_process);
             double num_flops = determineFlops(num_bytes, total_data_size);
             // WRENCH_INFO("Chunk: %.2lf bytes / %.2lf flops", num_bytes, num_flops);
             // Start the computation asynchronously
             simgrid::s4u::ExecPtr exec = simgrid::s4u::this_actor::exec_init(num_flops);
-            exec->start();
-            double exec_start_time = exec->get_start_time();
-            // Read data from the file
-            read_start_time = wrench::Simulation::getCurrentSimulatedDate();
-            fs.second->getStorageService()->readFile(fs.first, fs.second, num_bytes);
-            read_end_time = wrench::Simulation::getCurrentSimulatedDate();
-            // Wait for the computation to be done
-            exec->wait();
+            double exec_start_time = 0.0;
+            double exec_end_time = 0.0;
+            if(n_prefetch_blocks_counter < this->n_prefetch_blocks){
+                exec->start();
+                exec_start_time = exec->get_start_time();
+                // Read data from the file
+                read_start_time = wrench::Simulation::getCurrentSimulatedDate();
+                fs.second->getStorageService()->readFile(fs.first, fs.second, num_bytes);
+                read_end_time = wrench::Simulation::getCurrentSimulatedDate();
+                // Wait for the computation to be done
+                exec->wait();
```

Comment:
This should be a 1 to 1 copy of previous setup, except that it is now executed only for the configured number of blocks.

Reply:
Or do you mean this interacts badly with the phase without prefetching?

Reply:
I would have to think more about that then

Reply:
> This should be a 1 to 1 copy of previous setup, except that it is now executed only for the configured number of blocks.

Yes, and it works for just one prefetch block in the buffer, because you get something like `compute->read->wait` and once it is finished, the next cycle will start. But once there is more than one prefetch block, the next prefetch cycle would call the next compute overwriting the old one, where it actually needs to only read. So we need a further decoupling of the read and compute steps.

Reply:
But in this case, the previous code was not correct, too, right?

Reply:
(i.e. assuming we prefetch everything in parallel)